File: src/MixedIL.Example/System/ObjectExtensions.cs
using System.Reflection;
using MixedIL;

namespace System;

public static class ObjectExtensions
{
    /// <summary>
    /// Set the property <see cref="AssemblyKeyNameAttribute.KeyName"/> that doesn't have setter by its BackingField.
    /// </summary>
    /// <param name="attribute"></param>
    /// <param name="keyName"></param>
    [MixedIL]
    public static extern void SetKeyName(this AssemblyKeyNameAttribute attribute, string keyName);
}

File: src/MixedIL.Example.FSharp/System/ObjectExtensions.fs
namespace System

open MixedIL
open System.Runtime.CompilerServices
open System.Reflection

[<Extension>]
type ObjectExtensions =
    [<MixedIL>]
    [<Extension>]
    static member SetKeyName(attribute: AssemblyKeyNameAttribute, keyName: string) : unit =
        raise (NotImplementedException())
